configure

```diff
@@ -799,10 +809,6 @@ def configure_intl(o):
   # set the default
   if with_intl is None:
     with_intl = 'none'  # The default mode of Intl
-  # sanity check localelist
-  if options.with_icu_locales and (with_intl != 'small-icu'):
-    print 'Error: --with-icu-locales only makes sense with --with-intl=small-icu'
-    sys.exit(1)
```

Comment:
I think you're right. While looking at it the first time, I interpreted it as a boolean check (do we want icu?) which it obviously isn't.

I'm not super happy with the 100-or-so lines of icu in here; but I'll try to rewrite as little as possible to cater for my validation. Thanks for the catch.